core/src/main/scala/kafka/message/ByteBufferMessageSet.scala

```diff
@@ -23,28 +23,45 @@ import kafka.common.KafkaException
 import java.nio.ByteBuffer
 import java.nio.channels._
 import java.io._
-import java.util.concurrent.atomic.AtomicLong
+import java.util.concurrent.atomic.{AtomicInteger, AtomicLong}
+
+import org.apache.kafka.common.utils.Utils
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
 
 object ByteBufferMessageSet {
 
-  private def create(offsetCounter: AtomicLong, compressionCodec: CompressionCodec, messages: Message*): ByteBuffer = {
+  private def create(offsetAssignor: OffsetAssigner,
+                     compressionCodec: CompressionCodec,
+                     wrapperMessageTimestamp: Option[Long],
+                     timestampType: TimestampType,
+                     messages: Message*): ByteBuffer = {
```

Comment:
Should we assert that wrapperMessageTimestamp is only set if compression is on and timestampType is logAppend?